src/client/Client.cc

```diff
@@ -13026,6 +13026,11 @@ int Client::_link(Inode *in, Inode *dir, const char *newname, const UserPerm& pe
     return -EDQUOT;
   }
 
+  if (dir->worm.is_enable() || in->worm.is_enable()) {
+    ldout(cct, 8) << "_link" << "worm feature doesn't support link " << dendl;
+    return -EPERM;
+  }
```

Comment:
Why not?

Reply:
We expect all files in the worm directory to be controlled，but it hardlink is supported，if a non-worm file linked to a file under worm directory, it can't  controlled by worm.